junit5/src/main/java/cz/xtf/junit5/extensions/helpers/CustomResourceHelper.java

```diff
@@ -0,0 +1,82 @@
+package cz.xtf.junit5.extensions.helpers;
+
+import cz.xtf.core.openshift.OpenShifts;
+import io.fabric8.kubernetes.client.dsl.MixedOperation;
+import io.fabric8.kubernetes.client.dsl.NonNamespaceOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+import io.fabric8.openshift.api.model.operatorhub.v1.DoneableOperatorGroup;
+import io.fabric8.openshift.api.model.operatorhub.v1.OperatorGroup;
+import io.fabric8.openshift.api.model.operatorhub.v1.OperatorGroupList;
+import io.fabric8.openshift.api.model.operatorhub.v1alpha1.ClusterServiceVersion;
+import io.fabric8.openshift.api.model.operatorhub.v1alpha1.ClusterServiceVersionList;
+import io.fabric8.openshift.api.model.operatorhub.v1alpha1.DoneableClusterServiceVersion;
+import io.fabric8.openshift.api.model.operatorhub.v1alpha1.DoneableInstallPlan;
+import io.fabric8.openshift.api.model.operatorhub.v1alpha1.DoneableSubscription;
+import io.fabric8.openshift.api.model.operatorhub.v1alpha1.InstallPlan;
+import io.fabric8.openshift.api.model.operatorhub.v1alpha1.InstallPlanList;
+import io.fabric8.openshift.api.model.operatorhub.v1alpha1.Subscription;
+import io.fabric8.openshift.api.model.operatorhub.v1alpha1.SubscriptionList;
+
+public class CustomResourceHelper {
```

Comment:
Can you add JavaDoc?

Moreover, what operations the class supports looks like general stuff in `OpsnShift` class. Like `openShift.getBuildConfigs()`. Could Appsint, OCP TS or other project benefit from such abstraction? WDYT @simkam @Crumby ? And by an abstraction I mean something like `openShift.customResources().getClusterServiceVersionClient()`

Reply:
hmm I'm not sure. Likely it wouldn't help. Typying would be problematic. 

From you example: `getClusterServiceVersionClient` -> `ClusterServiceVersion` is in fabric8 client, but for other resources you have implement types yourself. Example from our operator tooling:

```
MixedOperation<WildFlyServer, WildFlyServerList, WildFlyServerDoneable, Resource<WildFlyServer, WildFlyServerDoneable>>
```

Reply:
@istraka I din't add javadoc on purpose because other classes usually don't have javadoc, shall I add it anyway?

Reply:
@istraka looking here https://github.com/fabric8io/kubernetes-client/issues/2784 I realized there is specific API for OLM resources:

```
OpenShiftClient osClient = new DefaultOpenShiftClient();
        ClusterServiceVersion maybeCSV = (ClusterServiceVersion) osClient.operatorHub().clusterServiceVersions().
                load(path.toString()).get();
        System.out.println(maybeCSV.getKind());
```

hence I removed class `CustomResourceHelper.java` and the need for javadoc

Reply:
Great! Thanks.